packages/suite/src/actions/suite/metadataLabelingActions.ts

```diff
@@ -529,6 +568,297 @@ export const addMetadata =
             return result.success;
         });
 
+/**
+ *
+ * @returns files in storage provider split into [[...current][...old][...renamedOld]].
+ */
+const getMetadataFiles =
+    () =>
+    async (dispatch: Dispatch, getState: GetState): Promise<[string[], string[], string[]]> => {
+        const providerInstance = dispatch(
+            metadataProviderActions.getProviderInstance({
+                clientId: selectSelectedProviderForLabels(getState())!.clientId,
+                dataType: 'labels',
+            }),
+        );
+
+        if (!providerInstance) {
+            throw new Error('no provider instance');
```

Comment:
thunks shall never throw, but `rejectWithValue` I think, right @juriczech ?

Reply:
It's possible to throw errors, the thunk will still result in a rejected state, but without added types from redux which the `rejectWithValue` adds. It's better to `rejectWithValue` when you want to listen to rejected thunks in other thunks / reducer. I believe it's better to always reject so that it's then easy to use everywhere.